Chap_API_Storage.tex

```diff
@@ -0,0 +1,225 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% Chapter: Storage support
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\chapter{Storage Support Definitions}
+\label{chap:api_storage}
+
+Distributed and parallel computing systems are increasingly embracing storage hierarchies to meet the diverse data management needs of applications and other systems software in a cost-effective manner.
+These hierarchies provide access to a number of distinct storage layers, with each potentially composed of different storage hardware (e.g., HDD, SSD, tape, PMEM), deployed at different locations (e.g., on-node, on-switch, on-site, WAN), and designed using different storage paradigms (e.g., file-based, object-based).
+Each of these systems offers unique performance and usage characteristics that storage system users should carefully consider to ensure the most efficient use of storage resources.
+
+PMIx enables users to better understand storage hierarchies by defining attributes that formalize storage system characteristics, state, and other parameters.
+These attributes can be queried by applications, I/O libraries and middleware, and workflow systems to discover available storage resources and to inform on which resources are most suitable for different I/O workload requirements.
+
+%%%%%%%%%%%
+\section{Storage support constants}
+
+\declarestruct{pmix_storage_type_t}
+
+The \refstruct{pmix_storage_type_t} structure is an enumerated type for specifying the type of storage system being used.
+
+\begin{constantdesc}
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_LUSTRE}
+Lustre storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_GPFS}
+GPFS (Spectrum Scale) storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_DATAWARP}
+Datawarp storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_NFS}
+NFS storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_CEPH}
+Ceph storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_EXT4}
+Ext4 storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_XFS}
+XFS storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_TMPFS}
+tmpfs storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_UNKNOWN_FS}
+Unknown file-based storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_DAOS}
+DAOS storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_S3}
+S3 storage system.
+%
+\declareconstitemNEW{PMIX_STORAGE_TYPE_GCP}
+GCP storage system.
```

Comment:
I'm not familiar with GCP.  Based on some googling and proximity to the S3 type, I'm assuming this is a Google Cloud storage offering?  Would it be worth it to extend the description a bit

Reply:
This makes sense to me, doesn't hurt to be more descriptive in the definition.